src/beamlime/dashboard/workflow_config_service.py

```diff
@@ -0,0 +1,130 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2025 Scipp contributors (https://github.com/scipp)
+"""
+Provides a protocol and an adapter for managing workflow configurations.
+
+This is not a real service but an adapter that wraps :py:class:`ConfigService` to
+make it compatible with the :py:class:`WorkflowConfigService` protocol. This simplifies
+the implementation and testing of :py:class:`WorkflowController`.
+"""
+
+from collections.abc import Callable
+from typing import Protocol
+
+from beamlime.config.models import ConfigKey
+from beamlime.config.workflow_spec import (
+    PersistentWorkflowConfigs,
+    WorkflowConfig,
+    WorkflowSpecs,
+    WorkflowStatus,
+)
+
+_persistent_configs_key = ConfigKey(
+    service_name='dashboard', key='persistent_workflow_configs'
+)
+
+
+class WorkflowConfigService(Protocol):
+    """Protocol for workflow controller dependencies."""
+
+    def get_persistent_configs(self) -> PersistentWorkflowConfigs:
+        """Get persistent workflow configurations."""
+        ...
+
+    def save_persistent_configs(self, configs: PersistentWorkflowConfigs) -> None:
+        """Save persistent workflow configurations."""
+        ...
+
+    def send_workflow_config(self, source_name: str, config: WorkflowConfig) -> None:
+        """Send workflow configuration to a source."""
+        ...
+
+    def subscribe_to_workflow_specs(
+        self, callback: Callable[[WorkflowSpecs], None]
+    ) -> None:
+        """Subscribe to workflow specs updates."""
+        ...
+
+    def subscribe_to_workflow_status(
+        self, source_name: str, callback: Callable[[WorkflowStatus], None]
+    ) -> None:
+        """Subscribe to workflow status updates for a source."""
+        ...
+
+
+class ConfigServiceAdapter(WorkflowConfigService):
+    """
+    Adapter to make ConfigService compatible with WorkflowConfigService protocol.
```

Comment:
`ConfigService` and `WorkflowConfigService` are in the same package. Why do we need an adapter between them instead of merging the two interfaces into one?

Reply:
Everything is in the same package, not a reason to put all in the same class?

Reply:
My question is mainly about whether there is a use for this extra layer of abstraction. Is `ConfigService` used through any other means that as a base class for `WorkflowConfigService`?

Reply:
Yes, it is used by all the `ConfigBackedParam` instances.